### ISSUE TITLE
respect new file location

### ISSUE THREAD

**Issue by kometenstaub** [tadashi-aikawa/obsidian-another-quick-switcher#58]
Currently, all newly created files get created in the vault root.

It would be great if the new file location setting could be respected.

```js
app.vault.getConfig('newFileLocation')
'root'
'current'
'folder' -> a user-specified folder
if ('folder') {
  app.vault.getConfig('newFileFolderPath')
  'folder-name'
}
```

**Comment by tadashi-aikawa** [comment]
Hi, @kometenstaub.

Various Complements already respect the following options.

![image](https://user-images.githubusercontent.com/9500018/174604616-907fad20-1015-45a8-bc8a-ca944742892d.png)

Does it satisfy your requirements?

**Comment by tadashi-aikawa** [comment]
Hi, @kometenstaub.

Another Quick Switcher already respect the following options.

![image](https://user-images.githubusercontent.com/9500018/174604616-907fad20-1015-45a8-bc8a-ca944742892d.png)

Does it satisfy your requirements?

**Comment by kometenstaub** [comment]
Huh, that's odd. That's the option I mean. 
For me it placed all the files at the vault root although I was in a folder. I'll try to repro.

**Comment by tadashi-aikawa** [comment]
If you can reproduce it in the sandbox vault, it will be very helpful for me :)

**Comment by kometenstaub** [comment]
Yes I can reproduce it in the sandbox vault with the same folder as current file option on Obsidian 0.15.2

**Comment by kometenstaub** [comment]
The code looks whether there is a folder in the query with regex, but it doesn't checkt the default file location setting.

https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/blob/5fea94cd2aadb480b23b125f0a0080e85bb1a976/src/app-helper.ts#L266-L278

**Comment by tadashi-aikawa** [comment]
Could you tell me the reproduction procedure? (especially the options)

**Comment by tadashi-aikawa** [comment]
Thanks. What are the following settings?

![](https://user-images.githubusercontent.com/9500018/174604616-907fad20-1015-45a8-bc8a-ca944742892d.png)

**Comment by kometenstaub** [comment]
![screenshot](https://user-images.githubusercontent.com/83140328/174613852-662e354d-343c-4162-a106-5e9b3a8c6090.png)

**Comment by tadashi-aikawa** [comment]
Released in [v5.0.2](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/5.0.2) 🚀

**Comment by tadashi-aikawa** [comment]
Ah.. I had mistaken 😅

**Comment by tadashi-aikawa** [comment]
OK. fixed in [v5.0.3](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/5.0.3).

**Comment by kometenstaub** [comment]
Thank you! :)
